app/src/main/java/org/groundplatform/android/ui/datacollection/tasks/polygon/DrawAreaTaskViewModel.kt

```diff
@@ -69,6 +69,10 @@ internal constructor(
   private val _draftArea: MutableStateFlow<Feature?> = MutableStateFlow(null)
   val draftArea: StateFlow<Feature?> = _draftArea.asStateFlow()
 
+  private var draftTag: Feature.Tag? = null
+  private val _draftUpdates = MutableSharedFlow<Feature>(extraBufferCapacity = 1)
```

Comment:
Do we really need this new flow? Can't we reuse `_draftArea` here as well?

Reply:
We introduced _draftUpdates because _draftArea is used for draft creation/clearing, while this flow handles high-frequency in-place updates during drawing.

Reusing _draftArea would cause the entire feature set to re-render on every small vertex move, leading to flicker and extra allocations.

_draftUpdates uses a SharedFlow with a small buffer to emit lightweight updates via tryEmit, keeping map rendering smooth without unnecessary churn.